presto-jdbc/src/main/java/com/facebook/presto/jdbc/QueryInterceptor.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.jdbc;
+
+import java.sql.Statement;
+import java.util.Map;
+
+public interface QueryInterceptor
+{
+    QueryInterceptor init(PrestoConnection connection, Map<String, String> properties);
+    void destroy();
+    PrestoResultSet preProcess(String sql, Statement interceptedStatement);
+    PrestoResultSet postProcess(String sql, Statement interceptedStatement, PrestoResultSet originalResultSet);
```

Comment:
We should add some documentation to these functions about the arguments and expected behavior since people in open source might want to implement these and not having to read the actual implementation will be helpful to quickly onboard people and move ahead with their business logic.

Reply:
I added some documentation to these methods